chore(all): enhancements in config and README files + fix commit linting functionality

christophercr opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Some enhancements:
- enabled/adapted some rules in the TSLint and codelyzer configs (they have been tested in Stark in this PR: https://github.com/NationalBankBelgium/stark/pull/1284 so no more changes will be needed in Stark once we use this package).
- minor corrections in the different `README` files

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

christophercr commented:
@SuperITMan as discussed, in this PR I've done some corrections. Specially these:

- `stylelint-config-prettier` is not a dependency anymore because it is not needed for our config. The developer can decide whether to use it or not but that doesn't affect our config. Moreover, **the `stylelint-config-prettier` and `tslint-config-prettier` should always be put in the last position inside the `extends` field of the config files!** See [here](https://github.com/prettier/tslint-config-prettier#get-started) and [here](https://github.com/prettier/stylelint-config-prettier#installation). Therefore we cannot use it internally in the configs of code-style package.
- the paths mentioned in the `README` files should contain the `/lib` folder to work. So I added it. Maybe we can move the folders that are under `/lib` one level up so they are accessible directly at the root so the path would indeed be: `"@nationalbankbelgium/code-style/tslint/4.5.x"`. What do you think?
- I added/adapted some rules in TSLint and codelyzer that I think are relevant and useful in Stark. So you can see that I also created this PR https://github.com/NationalBankBelgium/stark/pull/1284 in Stark with those rules and the fixes. So you can check there the benefits of those rules.

By the way, it seems I cannot add/remove reviewers/labels/projects to the PRs in this repo... can you give me the permissions? :wink:

SuperITMan commented on lib/codelyzer/README.md:
```diff
@@ -1,24 +1,24 @@
 The following versions are available (based on **codelyzer** version):
 
--   4.5.x - `@nationalbankbelgium/code-style/tslint-codelyzer/4.5.x`
+-   4.5.x - `@nationalbankbelgium/code-style/lib/codelyzer/4.5.x`
```
Small typo: It should be: `@nationalbankbelgium/code-style/codelyzer/4.5.x`

christophercr replied:
As I mentioned in my last comment, the paths should contain the `/lib` folder otherwise the config cannot be found. The reason is that we have all these configs under the `/lib` folder. Therefore I also mentioned that we could move the folders one level up so they are accessible directly at the root so the path would indeed be: "@nationalbankbelgium/code-style/codelyzer/4.5.x"

christophercr replied:
Never mind, I didn't test this properly :(

SuperITMan commented on lib/codelyzer/4.5.x/README.md:
````diff
@@ -14,7 +14,7 @@ Adapt the content of your `tslint.json` file as follows:
 ```json
 {
-	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/tslint-codelyzer/4.5.x"],
+	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/lib/codelyzer/4.5.x"],
````
Small typo: It should be: `@nationalbankbelgium/code-style/codelyzer/4.5.x`

SuperITMan commented on lib/codelyzer/README.md:
````diff
@@ -1,24 +1,24 @@
 ```json
 {
-	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/tslint-codelyzer/4.5.x"],
+	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/lib/codelyzer/4.5.x"],
````
Small typo: It should be: `@nationalbankbelgium/code-style/codelyzer/4.5.x`

SuperITMan commented on lib/prettier/1.16.x/README.md:
````diff
@@ -1,15 +1,15 @@
 
 ```js
-module.exports = require("@nationalbankbelgium/code-style/prettier");
+module.exports = require("@nationalbankbelgium/code-style/lib/prettier/1.16.x");
````
Small typo: It should be: `@nationalbankbelgium/code-style/prettier/1.16.x`

SuperITMan commented on lib/prettier/README.md:
```diff
@@ -8,12 +8,12 @@ This sub-package hosts the [Prettier](https://prettier.io) configuration.
 The following versions are available (based on **Prettier** version):
 
--   9.10.x - `@nationalbankbelgium/code-style/prettier/1.16.x`
+-   9.10.x - `@nationalbankbelgium/code-style/lib/prettier/1.16.x`
```
Small typo: It should be: `@nationalbankbelgium/code-style/prettier/1.16.x`

SuperITMan commented on lib/prettier/README.md:
````diff
@@ -8,12 +8,12 @@ This sub-package hosts the [Prettier](https://prettier.io) configuration.
 
 ```js
-module.exports = require("@nationalbankbelgium/code-style/prettier/1.16.x");
+module.exports = require("@nationalbankbelgium/code-style/lib/prettier/1.16.x");
````
Small typo: It should be: `@nationalbankbelgium/code-style/prettier/1.16.x`

SuperITMan commented on lib/stylelint/9.10.x/README.md:
````diff
@@ -6,15 +6,13 @@
 ```json
 {
-	"extends": "@nationalbankbelgium/code-style/stylelint/9.10.x",
+	"extends": "@nationalbankbelgium/code-style/lib/stylelint/9.10.x",
````
Small typo: It should be: `@nationalbankbelgium/code-style/stylelint/9.10.x`

SuperITMan commented on lib/tslint/README.md:
````diff
@@ -8,15 +8,15 @@ This sub-package hosts the [TSLint](https://palantir.github.io/tslint/) configur
 ```json
 {
-	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/tslint"],
+	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/lib/tslint/4.3.x"],
````
Small typo: It should be: `@nationalbankbelgium/code-style/tslint/4.3.x`

SuperITMan commented on lib/tslint/README.md:
```diff
@@ -8,15 +8,15 @@ This sub-package hosts the [TSLint](https://palantir.github.io/tslint/) configur
 The following versions are available (based on **TSLint** version):
 
--   3.1.x - `@nationalbankbelgium/code-style/tslint/4.3.x`
+-   4.3.x - `@nationalbankbelgium/code-style/lib/tslint/4.3.x`
```
Small typo: It should be: `@nationalbankbelgium/code-style/tslint/4.3.x`

SuperITMan commented on lib/tslint/4.3.x/README.md:
````diff
@@ -12,7 +12,7 @@ Adapt the content of your `tslint.json` file as follows:
 ```json
 {
-	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/tslint/4.3.x"],
+	"extends": ["tslint:latest", "whatever config", "@nationalbankbelgium/code-style/lib/tslint/4.3.x"],
````
Small typo: It should be: `@nationalbankbelgium/code-style/tslint/4.3.x`

SuperITMan commented on lib/tsconfig/README.md:
````diff
@@ -8,16 +8,16 @@ This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configu
 ```json
 {
-	"extends": "@nationalbankbelgium/code-style/tsconfig/3.1.x",
+	"extends": "@nationalbankbelgium/code-style/lib/tsconfig/3.1.x",
````
Small typo: It should be: `@nationalbankbelgium/code-style/tsconfig/3.1.x`

SuperITMan commented on lib/tsconfig/README.md:
```diff
@@ -8,16 +8,16 @@ This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configu
--   3.1.x - `@nationalbankbelgium/code-style/tsconfig/3.1.x`
--   3.1.x and NG 7 - `@nationalbankbelgium/code-style/tsconfig/3.1.x/ng7`
+-   3.1.x - `@nationalbankbelgium/code-style/lib/tsconfig/3.1.x`
+-   3.1.x and Angular 7 - `@nationalbankbelgium/code-style/lib/tsconfig/3.1.x/ng7`
```
Small typo: It should be: `@nationalbankbelgium/code-style/tsconfig/3.1.x/ng7`

SuperITMan commented on lib/tsconfig/README.md:
```diff
@@ -8,16 +8,16 @@ This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configu
 
--   3.1.x - `@nationalbankbelgium/code-style/tsconfig/3.1.x`
--   3.1.x and NG 7 - `@nationalbankbelgium/code-style/tsconfig/3.1.x/ng7`
+-   3.1.x - `@nationalbankbelgium/code-style/lib/tsconfig/3.1.x`
```
Small typo: It should be: `@nationalbankbelgium/code-style/tsconfig/3.1.x`

SuperITMan commented on lib/tsconfig/3.1.x/ng7/README.md:
````diff
@@ -12,7 +12,7 @@ Adapt the content of your `tsconfig.json` file as follows:
 ```json
 {
-	"extends": "@nationalbankbelgium/code-style/tsconfig/3.1.x/ng7",
+	"extends": "@nationalbankbelgium/code-style/lib/tsconfig/3.1.x/ng7",
````
Small typo: It should be: `@nationalbankbelgium/code-style/tsconfig/3.1.x/ng7`

SuperITMan commented on lib/tsconfig/3.1.x/README.md:
````diff
@@ -12,7 +12,7 @@ Adapt the content of your `tsconfig.json` file as follows:
 ```json
 {
-	"extends": "@nationalbankbelgium/code-style/tsconfig/3.1.x",
+	"extends": "@nationalbankbelgium/code-style/lib/tsconfig/3.1.x",
````
Small typo: It should be: `@nationalbankbelgium/code-style/tsconfig/3.1.x`

SuperITMan commented on lib/stylelint/README.md:
````diff
@@ -6,19 +6,17 @@
 ```json
 {
-	"extends": "@nationalbankbelgium/code-style/stylelint",
+	"extends": "@nationalbankbelgium/code-style/lib/stylelint/9.10.x",
````
Small typo: It should be: `@nationalbankbelgium/code-style/stylelint/9.10.x`

SuperITMan commented on lib/stylelint/README.md:
```diff
@@ -6,19 +6,17 @@
 The following versions are available (based on **stylelint** version):
 
--   9.10.x - `@nationalbankbelgium/code-style/stylelint/9.10.x`
+-   9.10.x - `@nationalbankbelgium/code-style/lib/stylelint/9.10.x`
```
Small typo: It should be: `@nationalbankbelgium/code-style/stylelint/9.10.x`

christophercr commented:
@SuperITMan I've updated this PR with the following changes:

- removed the `/lib` part of the path in the README files.
- created a new folder for TypesScript 3.2 and added the new rule [`strictBindCallApply`](https://github.com/microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#strictbindcallapply) there.
- added `husky` and `cz-customizable` as dependencies to fix the commit linting functionality (it was broken)
